.github/workflows/scripts/models/run_tp.sh

```diff
@@ -0,0 +1,188 @@
+#!/bin/bash
+set -eo pipefail
+set -x
+
+# IMPORTANT! we use half of one socket cores to simulate TP functionality
+cores_list=(24)
+# model_list=("llama2-7b" "llama2-13b" "llama2-70b" "gptj-6b" "baichuan-13b")
+model_list=("llama2-7b")
+input_list=(32 1024 2016)
+output=32
+beam_list=(1)
+precision_list=("q4_j")
+
+function main {
+    conda_env="$1"
+    # get the compiler version
+    gcc_version_info=$(gcc --version)
+    compiler_version=$(echo "$gcc_version_info" | grep -oP 'gcc \(.*\) \K[0-9]+\.[0-9]+\.[0-9]+' | head -1)
+
+    working_dir="${WORKING_DIR}"
+    scripts_dir="${working_dir}/neural_speed/convert"
+
+    # init conda
+    . $(dirname ${CONDA_EXE})/../etc/profile.d/conda.sh
+    conda activate $conda_env || source activate $conda_env
+    pip install cmake psutil
+    if [[ "${compiler_version}" != "12.3.0" ]]; then
+        conda install --update-deps -c conda-forge gxx==${compiler_version} gcc==${compiler_version} gxx_linux-64==${compiler_version} libstdcxx-ng sysroot_linux-64 -y
+    fi
+
+    # check oneCCL and build
+    ccl_dir=${working_dir}/oneCCL/build/_install
+    if [ ! -d "$ccl_dir" ]; then
+        cd ${working_dir}
+        git clone https://github.com/intel/oneCCL.git
+        cd ${working_dir}/oneCCL
+        git checkout 2021.9
+        sed -i 's/cpu_gpu_dpcpp/./g' cmake/templates/oneCCLConfig.cmake.in
+        mkdir build && cd build
+        cmake ..
+        make -j install
+    fi
```

Comment:
shall we check whether ccl has been installed to save time? for example, do we recommend to install oneCCL via oneAPI?

Reply:
oneCCL is part of oneAPI, once oneAPI installed, oneCCL can be activated via ```source /path/to/oneapi/setvars.sh```

Reply:
we're doing some work that could reduce the CI compilation time. If oneCCL can be linked from oneAPI, just use the pre-compiled library.

Reply:
I will prefer sourcing in this file because other parts of CI might not source oneapi

```
if [ -z ${ONEAPI_ROOT} ]; then
  if [ -d /opt/intel/oneapi ]; then
    ONEAPI_ROOT=/opt/intel/oneapi
  elif [ -d ~/intel/oneapi ]; then
    ONEAPI_ROOT=~/intel/oneapi
  fi
fi

if [ ! -z ${ONEAPI_ROOT} ]; then
  # source setvars.sh
  source ${ONEAPI_ROOT}/setvars.sh
else
  # build ONECCL from source
```